kyo-core/shared/src/main/scala/kyo/Resource.scala

```diff
@@ -114,7 +114,7 @@ object Resource:
                                 }
                             }
                                 .unit
-                                .pipe(Async.run)
+                                .pipe(Async.run[Nothing, Unit, Any])
```

Comment:
This is the example I mentioned in the description where the compiler picks up the isolate in the enclosing context, which requires the workaround with explicit type params